sync/src/chain.rs

```diff
@@ -2082,16 +2082,12 @@ impl ChainSync {
 
 		if !enacted.is_empty() {
 			// Select random peers to re-broadcast transactions to.
-			let mut random = random::new();
-			let len = self.peers.len();
-			let peers = random.gen_range(0, min(len, 3));
-			trace!(target: "sync", "Re-broadcasting transactions to {} random peers.", peers);
-
-			for _ in 0..peers {
-				let peer = random.gen_range(0, len);
-				self.peers.values_mut().nth(peer).map(|mut peer_info| {
+			let peers: Vec<PeerId> = ChainSync::select_random_peers(&self.peers.keys().cloned().collect::<Vec<_>>());
+			trace!(target: "sync", "Re-broadcasting transactions to random peers.");
+			for peer in peers.iter().take(3) {
```

Comment:
Its up to 3 peers, but it also maintains the `< sqrt(total_peers)`, since it uses the common method for selecting random peers. Feel free to change to what ever numbers you think are justified. The solution without allocation will sometimes select the same peers multiple times (common in private 2-3 peer networks), no? You can do `.take(random::new().get_rng(0, self.peers.len()))` to recover the less deterministic behaviour.